packages/testing-library/utils/test-cases.ts

```diff
@@ -160,8 +161,12 @@ export async function runTestCases(testCases: Map<string, TestCase[]>, testOptio
         },
     });
 
-    const rejectedAsyncPromisesResult = allAsyncResults.rejectedIndexes.map<AdderError>((x) => allAsyncResults.taskResults[x]!);
-    const rejectedSyncPromisesResult = allSyncResults.rejectedIndexes.map<AdderError>((x) => allSyncResults.taskResults[x]!);
+    const rejectedAsyncPromisesResult = allAsyncResults.rejectedIndexes.map<AdderError>(
+        (x) => allAsyncResults.taskResults[x] as unknown as AdderError,
+    );
+    const rejectedSyncPromisesResult = allSyncResults.rejectedIndexes.map<AdderError>(
+        (x) => allSyncResults.taskResults[x] as unknown as AdderError,
+    );
```

Comment:
i think this is another scenario where `strictTypeChecked` is being too restrictive again - preventing the type assertion with the `!` and forcing us to use the `as unknown as Type` hack

Reply:
There is an argument to be made about code clarity vs. readability, as you said. In this case I totally agree with the strict rule, if we do things that make typescript unhappy, we should at least be able to identify them quickly. 

Even when I got this error and the error message stated something about an exclamation mark, i read past it because it's so small. Let's keep it this way.